packages/demo/src/components/examples/TextInputExamples.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import {InputValidator, TextInput, TextInputProvider, Section} from 'react-vapor';
+
+const nonEmptyValidation: InputValidator = (value: string) => {
+    const isEmpty = !value;
+    const isWhiteSpace = !isEmpty && !value.trim();
+    if (isWhiteSpace) {
+        return {status: 'warning', message: 'A white space is not empty, but is not ideal.'};
+    } else if (isEmpty) {
+        return {status: 'invalid', message: 'Cannot be empty.'};
+    } else {
+        return {status: 'valid'};
+    }
+};
+
+export const TextInputExamples: React.FunctionComponent = () => (
```

Comment:
Maybe add an example that uses 2 inputs since it's very probable it will happen in a real scenario? :P 

And another example with a default value?

I'd also like to see how I can leverage the state with `useTextInput` to drive something else, a scenario like "This field is only shown when you have entered a value in the first one", or "You can't click Next if the input is empty"

Reply:
Added a hook usage demo in the example page, let me know if it doesn't cover something you think should be in there.

Reply:
Amazing! The example really clears things out

![image](https://user-images.githubusercontent.com/8355585/121704874-26d5f180-caa2-11eb-9a31-aabfa8f22c1c.png)